### ISSUE TITLE
update nodemailer configuration parameter for production

### ISSUE THREAD

**Issue by jocrah** [tbf-challenge/teamwork-be#176]
## Task Description

Based on some setup guides for `sendgrid` and `mailtrap` and so on, we might need to update the `service` parameter we use for nodemailer when in production. We will instead use the `host` and `port` like we do in development.

References
1. https://www.twilio.com/blog/send-smtp-emails-node-js-sendgrid
2. https://mailtrap.io/blog/sending-emails-with-nodemailer/ 

## Technical Considerations
Check the `src/lib/emailLib` and for the conditional which checks if the env is production before using the `service` argument.

## Additional Considerations
1. We can do this such that when `port` and `host` are defined, we use them, else if service is defined, we use the service.
2. Also, in light of this change, it will be good to have unified envs. it's fine to go with just the `MAIL_SERVICE_*`  envs for now without checking the `NODE_ENV`. Any dev that wants to test can still provide their own details or also have a dev account on sengrid and use that. 

It helps to ensure we have similar behaviour in the development and in production so there are no surprises its not hard to debug when some part isn't working

**Comment by jocrah** [comment]
This is an update on what I just learnt about how the `service` parameter (we used earlier) worked. For context, this is how the code looked like:

https://github.com/tbf-challenge/teamwork-be/blob/894a2d98918f065969e2caf2aec9621959b4b717/src/lib/email-lib.js#L7-L9


I should have read some more on this.  I wasn't seeing its usage that much while trying to read on it. That's why this issue was created.

And though the 'issue' has been resolved, it was still on my mind.

My curiosity led me to the nodemailer source code. Imagine my surprise when I found out the `service` parameter is basically nodemailer trying to help us so that we don't have to enter the `host` and `port` ourselves. 

That is,  they themselves have written down corresponding`host` and `port` values for the common services (gmail, sendgrid and a lot more)

This is [the file](https://github.com/nodemailer/nodemailer/blob/6f405365b5dd20a8403a3d25f6f4d91bf5dbf8fc/lib/well-known/services.json) in their source code that contains the list of services and their corresponding `host` and `port` configs.

So basically this issue was not needed, apologies team 😅 (esp. @itzgeebee and thanks too)

cc @jakazzy @beblicarl
